chain/network/src/peer_manager/tests/tier2.rs

```diff
@@ -0,0 +1,252 @@
+use crate::broadcast;
+use crate::network_protocol::testonly as data;
+use crate::peer_manager::connection_store::STORED_CONNECTIONS_MIN_DURATION;
+use crate::peer_manager::peer_manager_actor::Event as PME;
+use crate::peer_manager::peer_manager_actor::{
+    POLL_CONNECTION_STORE_INTERVAL, RECONNECT_ATTEMPT_INTERVAL, UPDATE_CONNECTION_STORE_INTERVAL,
+};
+use crate::peer_manager::testonly::start as start_pm;
+use crate::peer_manager::testonly::ActorHandler;
+use crate::peer_manager::testonly::Event;
+use crate::tcp;
+use crate::testonly::make_rng;
+use crate::time;
+use near_o11y::testonly::init_test_logger;
+use near_primitives::network::PeerId;
+use near_store::db::TestDB;
+use std::sync::Arc;
+
+async fn check_recent_outbound_connections(pm: &ActorHandler, mut wanted: Vec<PeerId>) {
+    let mut got: Vec<PeerId> = pm
+        .with_state(move |s| async move {
+            s.connection_store
+                .get_recent_outbound_connections()
+                .iter()
+                .map(|c| c.peer_info.id.clone())
+                .collect()
+        })
+        .await;
+
+    wanted.sort();
+    got.sort();
+
+    assert!(got == wanted, "Expected {:?} but got {:?}", wanted, got);
```

Comment:
assert_eq, it will display the mismatching values by default. Also consider using as_set(), instead of sorting.